src/util/system.cpp

```diff
@@ -310,8 +311,16 @@ bool ArgsManager::ParseParameters(int argc, const char* const argv[], std::strin
             key[0] = '-';
 #endif
 
-        if (key[0] != '-')
-            break;
+        if (key[0] != '-') {
+            if (accept_any_command) break;
+            Optional<unsigned int> flags = GetArgFlags(key);
+            if (!flags || !(*flags & ArgsManager::COMMAND)) {
+                error = strprintf("Invalid command '%s'", argv[i]);
+                return false;
+            }
+            m_commands.push_back(key);
+            continue;
```

Comment:
I'm not sure multiple (sub)commands makes sense for us. At present, I think we have:

 * bitcoind - doesn't take subcommands
 * bitcoin-wallet - wants a single subcommand
 * bitcoin-tx - wants multiple instructions with parameters
 * bitcoin-cli - wants to do its own parsing of stuff after the options
 * bitcoin-qt - does its own parsing of stuff after the options for payment server support
 
For bitcoin-util, I think we want it to be more like bitcoin-wallet, and if we were to do a psbt version of bitcoin-tx's multiple commands in order to construct a psbt, I think that would look like `bitcoin-util psbt <subsubcmd> <subsubcmd> ..`.

So I don't think it makes much sense to support multiple subcommands, particularly without letting them have their own params. Which I think means you could have a nicer implementation with something like https://github.com/ajtowns/bitcoin/commits/202101-getcommand which lets you just say `AddCmd(...)` to start introducing new commands, don't have to explicitly set `accept_any_command` when parsing, and can just invoke `GetCommand()` and `GetArguments()` to get the pre-parsed results. Thoughts?

I've added a second commit on that branch that allows the ArgsManager options to come after the command, which is the same as this PR, but differs from current master (which leaves later options up to the command -- so bitcoin-cli treats them as rpc args, while bitcoin-wallet currently just silently ignores them).

Reply:
Thanks, took most of your changes